src/pybind/mgr/cephadm/module.py

```diff
@@ -411,7 +411,7 @@ def __init__(self, *args: Any, **kwargs: Any):
 
         self.cephadm_binary_path = self._get_cephadm_binary_path()
 
-        self._worker_pool = multiprocessing.pool.ThreadPool(10)
+        self._worker_pool = ThreadPoolExecutor(max_workers=10)
```

Comment:
> fix multiprocessing pool deadlock by means of replacing multiprocessing.pool.map to ThreadPoolExecutor

again, i'd suggest to summarize the change in the title briefly, in this case, it'd be better to just put

> mgr/cephadm: replace multiprocessing.pool.ThreadPool with ThreadPoolExecutor
>
> * replace multiprocessing.pool.map with ThreadPoolExecutor.map
> * replace multiprocessing.pool.ThreadPool with ThreadPoolExecutor
>
> this change fixes multiprocessing pool deadlock
>
> /// more details here

also, better off referencing python document at https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing.dummy, in which it puts

> Note: A ThreadPool shares the same interface as Pool, which is designed around a pool of processes and predates the introduction of the concurrent.futures module. As such, it inherits some operations that don’t make sense for a pool backed by threads, and it has its own type for representing the status of asynchronous jobs, AsyncResult, that is not understood by any other libraries.
Users should generally prefer to use concurrent.futures.ThreadPoolExecutor, which has a simpler interface that was designed around threads from the start, and which returns concurrent.futures.Future instances that are compatible with many other libraries, including asyncio.

as i think the python document provides a more specific solution than the article at https://pythonspeed.com/articles/python-multiprocessing/  . also the former is likely to last longer over time.